config/v1/types_build.go

```diff
@@ -29,6 +29,10 @@ type BuildSpec struct {
 	// BuildOverrides controls override settings for builds
 	// +optional
 	BuildOverrides BuildOverrides `json:"buildOverrides,omitempty"`
+	// RegistriesConfig is a reference to a ConfigMap containing the registry configuration
+	// for image search, allowed insecure registries, and blocked registries.
+	// +optional
+	RegistriesConfig ConfigMapReference `json:"registriesConfig,omitempty"`
```

Comment:
i'm torn over whether this should be top level, or put in BuildDefaults so that we can later allow end-users to provide their own value if they want to....  i'm leaning towards the latter for flexibility, even if we decide we never want to expose t.

Similarly, while you're in here, can you add HTTP/HTTPS/NOPROXY settings to BuildDefaults?  (Same logic.. we may or may not want to let end-users specify their own proxy settings on a per build basis in the future).

Reply:
(we may also want to revisit moving additionalTrustedCA into BuildDefaults for the same reason, but for now i'm content to leave it while we get things working)

Reply:
I had similar thoughts about the HTTP proxies - these should be first-class in some capacity, rather than buried in an ENV var. I'll add those.

Adding to build defaults does give us flexibility. In the long run I'd be worried about security concerns/pushback for such a feature. At present, though, I don't think there is harm going with this approach.

Reply:
I'm fine with it, though i think @deads2k has been against inlining content like that historically, i can't remember the reasoning though.  It certainly would make the consuming logic a lot simpler.